mesonbuild/interpreter/interpreterobjects.py

```diff
@@ -809,6 +809,11 @@ def get_exe(self) -> T.Union[ExternalProgram, build.Executable, build.CustomTarg
     def get_name(self) -> str:
         return self.name
 
+
+class Doctest(Test):
+    target: build.BuildTarget
```

Comment:
This isn't a dataclass, what are you expecting to see here?

Reply:
I thought it'd let mypy know the type? The `target` is always written after creation, so you don't need to override `__init__`.

Reply:
that tells mypy that this field always exists and is always a build.BuildTarget. I guess it works, even though it's not always true.

Reply:
True, it has to be Optional and `= None`.